pkg/arvo/neo/cod/std/src/con/task-htmx.hoon

```diff
@@ -1,276 +1,361 @@
-        =/  that  -
-        =/  classes
-          %+  weld
-            "b0 br1 hover p1 grow tl action mono fr g3"
-          ?:(done.t " strike f3" "")
-        that(a.g [[%class classes] a.g.that])
-      ^-  manx
-      ;button
-        =type  "button"
-        =onclick
-          """
-          this.classList.toggle('toggled'); this.parentNode.nextElementSibling.classList.toggle('hidden');
-          """
-        ;span.bold: {(trip -:(pout pith))}
-        ;span.break.f2: {(scag (fall (find [10 ~] (trip text.t)) 30) (trip text.t))}
+  ::  <div class="fc g1" here="{<pt>}" done="" ...>
+  ::  toggle hidden attribute on buttons-menu div
+  ;div.fr.g1.p1
+    =here  pt
+    =onmouseover  "this.childNodes[1].classList.remove('hidden');$(this).addClass('b1 br2');"
+    =onmouseout   "this.childNodes[1].classList.add('hidden');$(this).removeClass('b1 br2');"
+      ::
+      ::  div with %edit poke functionality
+      ::  div that sends %edit poke when input with class="text"
+      ::  or input with class "done" are being changed
+      ;div.fr.ac.g1.grow
+        =hx-post     "/neo/hawk{pt}?stud=task-diff"
+        =hx-trigger  "input changed delay:0.4s from:find .text, input from:find .done"
+        =hx-swap     "none"
+        =head        "edit"
+          ;+
+          ::  defines class attribute with class names
+          =/  class  [%class "p2 br1 border done s3"]
+          ::
+          ::  checkbox logic:
+          ::  - if task is toggled, checkbox will
+          ::    appear as checked
+          ::  - if task has kids and all kids are done,
+          ::    user will be free to toggle the task
+          ::  - if task have kids and they are not done,
+          ::    checkbox will have readonly attribute and
+          ::    will show alert onclick
+          ::  - even though we have logic for handling
+          ::    toggling of a parent task, we prevent
+          ::    a task from being marked as done if it
+          ::    has untoggled kids
+          ::
+          ::  combining attribute logic with manx below
+          =;  m
+            ::  checks if the task toggled as done
+            ?.  done.t
+              ::  if it's not done, does it have kids?
+              ?~  order.t
+                ::
+                ::  a.g.m is a part of the manx that
+                ::  contains input checkbox attributes;
+                ::  we assign the class attribute to
+                ::  the rest of its data
+                m(a.g [class a.g.m])
+              =/  kc
+                (kids-check pith)
+              ~&  >>>  kc
+              ?:  kc
+                ::  assigning class attribute to
+                ::  the rest of manx data
+                m(a.g [class a.g.m])
+              ::
+              ::  assigns readonly, id and class
+              ::  attributes to checkbox; id will trigger
+              ::  alert script functionality
+              m(a.g [[%readonly ""] [%id "alert"] class a.g.m])
```

Comment:
You can see from the three sigpams I added that the new kids-check logic is accurately tracking whether or not the kids are done, but for whatever reason, this class is only ever being added to the first element whose kids aren't done. @YunPearce